pkg/apis/pipeline/v1beta1/when_types.go

```diff
@@ -99,9 +104,11 @@ type WhenExpressions []WhenExpression
 // AllowsExecution evaluates an Input's relationship to an array of Values, based on the Operator,
 // to determine whether all the When Expressions are True. If they are all True, the guarded Task is
 // executed, otherwise it is skipped.
-func (wes WhenExpressions) AllowsExecution() bool {
+// If CEL expression exists, AllowsExecution will get the evaluated results from evaluatedCEL and determine
+// if the Task should be skipped.
+func (wes WhenExpressions) AllowsExecution(evaluatedCEL map[string]bool) bool {
 	for _, we := range wes {
-		if !we.isTrue() {
+		if !we.isTrue(evaluatedCEL) {
```

Comment:
I don't think we should pass in the full map here, maybe we can evaluate it right here then instead?

```suggestion
                if we.CEL != "" {
                    return evaluatedCEL[we.CEL]
	        }
		if !we.isTrue() {
```